torchtitan/parallelisms/parallelize_llama.py

```diff
@@ -350,37 +350,41 @@ def apply_tp(model, world_mesh, parallel_dims, job_config: JobConfig):
         {
             "tok_embeddings": RowwiseParallel(
                 input_layouts=Replicate(),
-                output_layouts=Shard(1),
```

Comment:
curious: could this be `output_layouts=Shard(0)` and then do not need the `PrepareModuleInput`?

Reply:
@awgu 
Currently we are doing folding after embedding layer, so we can't do what you suggested.
But I just realize that maybe we can do folding even before embedding layer, then I think we can do this, just like the non-folding case.

Reply:
@awgu 
OK I tried out the change. Please see comparison [here](https://github.com/pytorch/torchtitan/pull/437/files/33160f9abd47aa2133191e63e9e7849ad8a93064..346038bcd18b71befc25db4b957c1f5567f8dacb).
Everything works except the CI [failure](https://github.com/pytorch/torchtitan/actions/runs/9789630903/job/27228806203) says
> RuntimeError: It seems that we cannot capture your model as a full graph. Typical reasons include graph breaks, data/shape-dependent control flow, or missing meta kernels for custom operators. You can use our manual pipeline interfaces, or try to fix the graph breaks

So I decided to change it back.